pkgs/applications/misc/onboard/default.nix

```diff
@@ -0,0 +1,155 @@
+  doCheck = false;
+  checkInputs = [
+    python3.pkgs.nose
+    customHunspell
+    (aspellWithDicts (dicts: with dicts; [ en ]))
+  ];
+
+  propagatedBuildInputs = [
+    python3
+    python3.pkgs.pycairo
+    python3.pkgs.dbus-python
+    python3.pkgs.pygobject3
+    python3.pkgs.systemd
+    python3.pkgs.distutils_extra
+    python3.pkgs.pyatspi
+    glib
+  ];
+
+  buildInputs = [
+    gtk3
+    libcanberra_gtk3
+    libudev
+    bash
+    hunspell
+    isocodes
+    xorg.libXtst
+    xorg.libxkbfile
+    libxkbcommon
+    gsettings_desktop_schemas
+    gnome3.dconf
+    wrapGAppsHook
+  ] ++ stdenv.lib.optional atspiSupport at_spi2_core;
+
+  nativeBuildInputs = [
+    pkgconfig
+    intltool
+    glibcLocales
+  ];
+
+  propagatedUserEnvPkgs = [
+    gnome3.dconf
+  ];
+
+  preBuild = ''
+    # Unnecessary file, has been removed upstream
+    # https://github.com/NixOS/nixpkgs/pull/24986#issuecomment-296114062
+    rm -r Onboard/pypredict/attic
+
+    substituteInPlace  ./scripts/sokSettings.py \
+      --replace "#!/usr/bin/python3" "" \
+      --replace "PYTHON_EXECUTABLE," "\"$out/bin/onboard-settings\"" \
+      --replace '"-cfrom Onboard.settings import Settings\ns = Settings(False)"' ""
+
+    chmod -x ./scripts/sokSettings.py
+
+    patchShebangs .
+
+    substituteInPlace  ./Onboard/LanguageSupport.py \
+      --replace "/usr/share/xml/iso-codes" "${isocodes}/share/xml/iso-codes" \
+      --replace "/usr/bin/yelp" "${yelp}/bin/yelp"
+
+    substituteInPlace  ./Onboard/Indicator.py \
+      --replace   "/usr/bin/yelp" "${yelp}/bin/yelp"
+
+    substituteInPlace  ./gnome/Onboard_Indicator@onboard.org/extension.js \
+      --replace "/usr/bin/yelp" "${yelp}/bin/yelp"
+
+    substituteInPlace  ./Onboard/SpellChecker.py \
+      --replace "/usr/share/hunspell" hunspell \
+      --replace "/usr/lib" "$out/lib"
+
+    substituteInPlace  ./data/org.onboard.Onboard.service  \
+      --replace "/usr/bin" "$out/bin"
+
+    substituteInPlace  ./Onboard/utils.py \
+      --replace "/usr/share" "$out/share"
+    substituteInPlace  ./onboard-defaults.conf.example \
+      --replace "/usr/share" "$out/share"
+    substituteInPlace  ./Onboard/Config.py \
+      --replace "/usr/share/onboard" "$out/share/onboard"
+
+    substituteInPlace  ./Onboard/WordSuggestions.py \
+      --replace "/usr/bin" "$out/bin"
+
+    # killall is dangerous on non-gnu platforms. Use pkill instead.
+    substituteInPlace  ./setup.py \
+      --replace "/bin/bash" ${stdenv.shell} \
+      --replace '"killall",' '"${procps}/bin/pkill", "-x",'
+  '';
+
+  postInstall = ''
+    cp onboard-default-settings.gschema.override.example $out/share/glib-2.0/schemas/10_onboard-default-settings.gschema.override
+
+    ${glib.dev}/bin/glib-compile-schemas $out/share/glib-2.0/schemas/
+  '';
+
+  meta = {
+    homepage = https://launchpad.net/onboard;
+    description = "An onscreen keyboard useful for tablet PC users and for mobility impaired users.";
+    longDescription = ''
```

Comment:
I would get rid of the long description. dconf is now added automatically thanks to `propagatedUserEnvPkgs`, changing advanced configuration is not NixOS specific, the a11y bus is not specific to this package… The only thing that matters is hunspell and we are do not even know how to use it.